File: 1264-maximum-number-of-words-you-can-type/1264-maximum-number-of-words-you-can-type.py
class Solution(object):
    def canBeTypedWords(self, text, brokenLetters):
        """
        :type text: str
        :type brokenLetters: str
        :rtype: int
        """
        words = text.split()
        wordCount = 0

        broken_set = set(brokenLetters)

        for word in words:
            if not any(letter in broken_set for letter in word):
                wordCount += 1

        return wordCount

File: 1264-maximum-number-of-words-you-can-type/1264-maximum-number-of-words-you-can-type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        unordered_set<char> letters(brokenLetters.begin(), brokenLetters.end()); 
        bool canType = true; 
        int result = 0;

        for(auto ch : text) {
            if(ch == ' ') {
                if(canType) {
                    result++;
                }
                canType = true;
            }
            else if(letters.find(ch) != letters.end()) {
                canType = false;
            }
        }
        if(canType) {
            result++;
        }
        return result;
    }
};